voter/src/main/java/org/terracotta/voter/ActiveVoter.java

```diff
@@ -77,15 +80,20 @@ public ActiveVoter start() {
 
   private Thread voterThread(CompletableFuture<VoterStatus> voterStatus, Optional<Properties> connectionProps, String... hostPorts) {
     return new Thread(() -> {
-      ScheduledExecutorService executorService = Executors.newScheduledThreadPool(hostPorts.length);
-      Set<String> registrationLatch = new HashSet<>();
-      registrationLatch.addAll(Arrays.asList(hostPorts));
-      List<ClientVoterManager> voterManagers = Stream.of(hostPorts).map(ClientVoterManagerImpl::new).collect(toList());
+      List<ClientVoterManager> clientVoterManagers = Stream.of(hostPorts).map(ClientVoterManagerImpl::new).collect(toList());
```

Comment:
As discussed, to make this class unit testable, lets pass a factory to the constructor to create a `ClientVoterManager`s instead of directly using `ClientVoterManagerImpl::new`